tasks/deploy/lib/environment-extensions.ts

```diff
@@ -132,12 +132,29 @@ extendEnvironment((hre) => {
                 ArtifactsMap[typeof contract]["abi"]
             >;
         }
+
         console.log(`deploying ${name}...`);
 
+        // Deploy libraries if necessary and link them to the target bytecode.
+        let artifact = await hre.artifacts.readArtifact(contract);
+        let libraries: Record<string, `0x${string}`> = {};
+        for (let libraryFilename in artifact.linkReferences) {
+            for (let libraryName in artifact.linkReferences[
+                libraryFilename as keyof typeof artifact.linkReferences
+            ] as any) {
```

Comment:
This code is a bit confusing to me. Why do we need to loop over this? Are there going to be multiple library names per library filename? Maybe I'm missing something.

Reply:
That's just how it's structured, unsure of a case where there would be multiple but figured looping again was the safe approach.

Here's an example from artifacts:

```json
"linkReferences": {
    "contracts/src/instances/morpho-blue/MorphoBlueConversions.sol": {
      "MorphoBlueConversions": [
        {
          "length": 20,
          "start": 16159
        },
        {
          "length": 20,
          "start": 16562
        }
      ]
    },
    "contracts/src/libraries/LPMath.sol": {
      "LPMath": [
        {
          "length": 20,
          "start": 7689
        }
      ]
    }
  }
  ```

Reply:
Ah I see. Makes sense since it's a `[string]: Object` type. Carry on 🫡